### PR TITLE
Add cask for HandBrake nightly (version :latest)

### PR DESCRIPTION
**Pull request by 500pd**
Adds a new cask for the nightly version of HandBrake

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-versions/pulls
[closed issues]: https://github.com/caskroom/homebrew-versions/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/handbrake-nightly.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cask 'handbrake-nightly' do
+  version :latest
+  sha256 :no_check
+
+  url do
+    require 'open-uri'
+    base_url = 'https://handbrake.fr/nightly.php'
+    file = URI(base_url).open.read.scan(%r{href="([^"]+.dmg)"}).flatten.first
+    file.to_s
```
Comment (reitermarkus) [inline Casks/handbrake-nightly.rb]: Remove this line.

Comment (500pd) [reply Casks/handbrake-nightly.rb]: The `file.to_s` line? If I remove that, `brew cask style --fix {{cask-file}}` says:
````
== Casks/handbrake-nightly.rb ==
W:  8:  5: Lint/UselessAssignment: Useless assignment to variable - file.

1 file inspected, 1 offense detected
Error: style check failed
````
`brew cask audit --download {{cask-file}}`, `brew cask install {{cask-file}}` and `brew cask uninstall {{cask-file}}` all work fine though.

Comment (reitermarkus) [reply Casks/handbrake-nightly.rb]: Yes, also remove the `file = ` assignment.

Comment (500pd) [reply Casks/handbrake-nightly.rb]: By removing both the `file.to_s` line and the `file =` line, `brew cask style --fix handbrake-nightly` fails with:
````
== Casks/handbrake-nightly.rb ==
W:  7:  5: Lint/UselessAssignment: Useless assignment to variable - base_url.

1 file inspected, 1 offense detected
Error: style check failed
````
`brew cask audit --download handbrake-nightly` doesn’t come with an error, but doesn't download the correct file.
`brew cask install handbrake-nightly` fails with:
````
Updating Homebrew...
==> Satisfying dependencies
==> Downloading https://handbrake.fr/nightly.php
######################################################################## 100.0%
==> No checksum defined for Cask handbrake-nightly, skipping verification
==> Installing Cask handbrake-nightly
==> Purging files for version latest of Cask handbrake-nightly
Error: Uh oh, could not figure out how to unpack '/Users/paad/Library/Caches/Homebrew/Cask/handbrake-nightly--latest.php'
````

Comment (reitermarkus) [reply Casks/handbrake-nightly.rb]: Only remove the `file =` part, not the whole line, sorry if that wasn't clear.

```ruby
base_url = 'https://handbrake.fr/nightly.php'
URI(base_url).open.read.scan(%r{href="([^"]+.dmg)"}).flatten.first
```

Comment (500pd) [reply Casks/handbrake-nightly.rb]: Ah. Now both `brew cask style --fix handbrake-nightly` and `brew cask audit --download handbrake-nightly` work. I updated the file and updated the pull request.

**Comment by reitermarkus** [comment]
Thanks!

**Pull request merged**